LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/ConcentratorDeduplication.cs

```diff
@@ -29,18 +31,18 @@ internal sealed class ConcentratorDeduplication : IConcentratorDeduplication, ID
         public ConcentratorDeduplication(
             IMemoryCache cache,
             WebSocketWriterRegistry<StationEui, string> socketRegistry,
-            ILogger<IConcentratorDeduplication> logger)
+            ILogger<IConcentratorDeduplication<T>> logger)
         {
             this.cache = cache;
             this.socketRegistry = socketRegistry;
             this.logger = logger;
         }
 
-        public bool ShouldDrop(UpstreamDataFrame updf, StationEui stationEui)
+        public bool ShouldDrop(T frame, StationEui stationEui)
```

Comment:
[nit] Shall we use `message` here as well like you do in `CreateCacheKey()`?

Reply:
Actually, I think frame is better suited here as `UpstreamDataFrame` and `JoinRequestFrame` both have frame in their names. Refactored to use frame everywhere (except for log messages where we should be user-friendlier) in d451eb10cceaad924c92e99b4ae88d5141b668f4